lib/matplotlib/backends/backend_svg.py

```diff
@@ -813,75 +815,57 @@ def draw_image(self, gc, x, y, im, dx=None, dy=None, transform=None):
             # to the clip-path
             self.writer.start('g', attrib={'clip-path': 'url(#%s)' % clipid})
 
-        trans = [1,0,0,1,0,0]
-        if rcParams['svg.image_noscale']:
-            trans = list(im.get_matrix())
-            trans[5] = -trans[5]
-            attrib['transform'] = generate_transform([('matrix', tuple(trans))])
-            assert trans[1] == 0
-            assert trans[2] == 0
-            numrows, numcols = im.get_size()
-            im.reset_matrix()
-            im.set_interpolation(0)
-            im.resize(numcols, numrows)
-
-        h,w = im.get_size_out()
-
-        if dx is None:
-            w = 72.0*w/self.image_dpi
-        else:
-            w = dx
-
-        if dy is None:
-            h = 72.0*h/self.image_dpi
-        else:
-            h = dy
-
-        oid = getattr(im, '_gid', None)
-        url = getattr(im, '_url', None)
+        oid = gc.get_gid()
+        url = gc.get_url()
         if url is not None:
             self.writer.start('a', attrib={'xlink:href': url})
         if rcParams['svg.image_inline']:
             bytesio = io.BytesIO()
-            _png.write_png(np.array(im)[::-1], bytesio)
+            _png.write_png(im, bytesio)
             oid = oid or self._make_id('image', bytesio.getvalue())
             attrib['xlink:href'] = (
                 "data:image/png;base64,\n" +
                 base64.b64encode(bytesio.getvalue()).decode('ascii'))
         else:
-            self._imaged[self.basename] = self._imaged.get(self.basename,0) + 1
+            self._imaged[self.basename] = self._imaged.get(self.basename, 0) + 1
             filename = '%s.image%d.png'%(self.basename, self._imaged[self.basename])
-            verbose.report( 'Writing image file for inclusion: %s' % filename)
-            _png.write_png(np.array(im)[::-1], filename)
+            verbose.report('Writing image file for inclusion: %s' % filename)
+            _png.write_png(im, filename)
             oid = oid or 'Im_' + self._make_id('image', filename)
             attrib['xlink:href'] = filename
 
-        alpha = gc.get_alpha()
-        if alpha != 1.0:
-            attrib['opacity'] = short_float_fmt(alpha)
-
         attrib['id'] = oid
 
         if transform is None:
+            w = 72.0 * w / self.image_dpi
+            h = 72.0 * h / self.image_dpi
+
             self.writer.element(
                 'image',
-                x=short_float_fmt(x/trans[0]),
-                y=short_float_fmt((self.height-y)/trans[3]-h),
+                transform = generate_transform([
+                    ('scale', (1, -1)), ('translate', (0, -h))]),
+                x=short_float_fmt(x),
+                y=short_float_fmt(-(self.height - y - h)),
```

Comment:
The transform is flipping it in place.  The math here is to convert the bottom location (which is used internally by matplotlib everywhere where increasing is "up") to the top location (which is used by most backends where increasing is "down").